frontend/components/UserForm/SingleDate.vue

```diff
@@ -4,94 +4,87 @@ v-model:      日付
 placeholder:  入力フォームに表示するプレイスホルダー
 -->
 <template>
-  <v-dialog
-    ref="dialog"
-    v-model="dialog"
-    persistent
-    width="290px"
-  >
-    <template v-slot:activator="{ on, attrs }">
-      <v-text-field
-        :value="newDate ? newDate : date"
-        single-line
-        readonly
-        outlined
-        append-icon="mdi-calendar-month"
-        v-bind="attrs"
-        v-on="on"
-        :placeholder="placeholder"
-        class="mt-1"
-      ></v-text-field>
-    </template>
-    <v-date-picker
-      v-model="inputDate"
-      scrollable
-      locale="jp-ja"
-      :day-format="date => new Date(date).getDate()"
-      color="primary"
-    >
-      <v-spacer></v-spacer>
-      <v-btn
-        text
+  <div>
+    <v-dialog v-model="dialog" persistent width="300px">
+      <template v-slot:activator="{ on, attrs }">
+        <v-text-field
+          v-model="date"
+          single-line
+          readonly
+          outlined
+          append-icon="mdi-calendar-month"
+          v-bind="attrs"
+          v-on="on"
+          @click="edit"
+          :placeholder="placeholder"
+          class="mt-1"
+        ></v-text-field>
+        <v-snackbar v-model="snackbar"> 表示日は過去にできません </v-snackbar>
+      </template>
+      <v-date-picker
+        v-model="editDate"
+        scrollable
+        locale="jp-ja"
+        :day-format="(date) => new Date(date).getDate()"
```

Comment:
無効なものを選んだらダイアログで示すのではなく、そもそも選べないようにするのはどうでしょうか？
例えばこんな風にです。
```suggestion
      <v-date-picker
        v-model="editDate"
        scrollable
        locale="jp-ja"
        :min="new Date().toISOString().substr(0, 10)"
        :day-format="(date) => new Date(date).getDate()"
```

Reply:
一回試してみます。